rfcs/0000_monty_versioning.md

```diff
@@ -0,0 +1,120 @@
+
+Using GitHub Action to increment the version number was prototyped and worked as expected. However, due to the constraints that Pull Requests are required, it turned out to be no more convenient than manually updating the version number via a normal Pull Request. The GitHub Action updated the version, created a Pull Request, and automatically merged it without approval. A normal Pull Request leaves the same commit and pull request audit trail, but requires Maintainer approval. Maintainer approval for updating the version number seems like a useful sanity check. In the future, when releases are triggered by version change, this approval will be helpful.
+
+```yaml
+name: Version
+
+permissions:
+  contents: write
+  pull-requests: write
+
+on:
+  workflow_dispatch:
+    inputs:
+      email:
+        description: "Email to use for the commit"
+        required: true
+        type: string
+      name:
+        description: "Name to use for the commit"
+        required: true
+        type: string
+      version:
+        description: "Version to set"
+        required: true
+        type: string
+
+jobs:
+  version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          lfs: true
+      - id: branch
+        run: |
+          echo "name=version-${{ inputs.version }}-$(date -u +"%Y%m%dT%H%M%SZ")" >> $GITHUB_OUTPUT
+      - run: |
+          git checkout -b ${{ steps.branch.outputs.name }}
+      - run: |
+          sed -i "s/__version__.*/__version__ = \"${{ inputs.version }}\"/" src/prototype/project/__init__.py
+      - run: |
+          git add src/prototype/project/__init__.py
+      - run: |
+          git config --global user.email "${{ inputs.email }}"
+      - run: |
+          git config --global user.name "${{ inputs.name }}"
+      - run: |
+          git commit -m "ci: version ${{ inputs.version }}"
+      - run: |
+          git push origin ${{ steps.branch.outputs.name }}
+      - run: |
+          gh pr create --base main --head ${{ steps.branch.outputs.name }} --title "ci: version ${{ inputs.version }}" --body "Version ${{ inputs.version }} set by @${{ github.event.sender.login }}"
+        env:
+          GH_TOKEN: ${{ github.token }}
+      - run: |
+          gh pr merge --admin --squash --delete-branch
+        env:
+          GH_TOKEN: ${{ github.token }}
+      - if: ${{ failure() }}
+        run: |
+          git push origin --delete ${{ steps.branch.outputs.name }}
+```
+
+# References
+
+* [Semantic Versioning](https://semver.org/) is a standard for versioning software.
+* [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) is a standard for commit messages.
+
+# Future possibilities
+
+If we were to adopt the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#specification) then the version number updates could be automated based on the commit message.
```

Comment:
We could also put this in the `tbp.cli` tool which would take care of most of the manual steps.

```
tbp bump-minor
```